### ISSUE TITLE
Feature request: generalizing the random.multivariate_* functions

### ISSUE THREAD

**Issue by vandenheuvel** [numpy/numpy#15530]
I think it would be practical to allow a generalized usage of the multivariate_* functions.

Currently, it is possible to use the `np.random.normal` function with multiple parameters at once:
```python
means, stds = np.array([0, 1e2]), np.array([1e-1, 1e-3])
assert(stds.shape[0] == means.shape[0])
x = np.random.normal(means, stds, size=(5, means.shape[0]))
print(x, x.shape)
```
giving for example
```
[[-4.10286750e-02  1.00000498e+02]
 [ 1.06635675e-01  1.00000462e+02]
 [ 1.03887646e-01  1.00001344e+02]
 [ 4.45630787e-02  9.99997015e+01]
 [ 1.22726902e-01  1.00001742e+02]] (5, 2)
```

I think this could be useful:
```python
dim = 2
multivariate_means = np.tile(means, (dim, 1))
multivariate_stds = np.tile(np.eye(dim), (dim, 1, 1))
assert(multivariate_means.shape[0] == multivariate_stds.shape[0])
x = np.random.multivariate_normal(means, stds, size=(5, means.shape[0], dim))
print(x, x.shape)
```
giving for example an array of shape ```(5, 2, 3)```.

Summarized: the input arguments of the multivariate_* could perhaps allow inputs to be of one dimension higher.

**Comment by seberg** [comment]
I am not sure I follow, It seems completely fine to me to allow broadcasting, e.g. make the signature `mean : (..., N)` and `cov : (..., N, N)` (where `...` is broadcast). I am not quite sure how I think size should behave (include `dim` or exclude it, I think probably exclude it actually).

I am not certain I understand your `multivariant_means`, since you *append* an extra dimension, but broadcasting rules always have the additional dimensions first. Disregarding your example though. Generalizations to higher dimensions are generally acceptable improvements, so PRs welcome.

**Comment by rossbar** [comment]
@vandenheuvel I'm not sure I quite understand either. `np.random.normal` already accepts n-dimensional input and as @seberg notes, is valid. For example, the desired shape from your second example using the inputs from your first example:

```python
>>> # From original example
>>> means, stds = np.array([0, 1e2]), np.array([1e-1, 1e-3])
>>> # Produce an array of shape (5, 2, 3)
>>> x = np.random.normal(means[:, np.newaxis], std[:, np.newaxis], size=(5, 2, 3))
>>> print(x.shape)
(5, 2, 3)
```
Does this example not accomplish what you had in mind? If not, can you use the example to explain how the behavior you had in mind differs from the given output?

**Comment by vandenheuvel** [comment]
@rossbar You produced independent realizations of one-dimensional random variables. Instead, I would like to create independent realizations of higher-dimensional random variables (where the different dimensions are not independent).

Think for example of creating a noisy swiss roll in higher dimensions.

**Comment by rossbar** [comment]
Sorry, I had originally missed that you were using `normal` in the top example but `multivariate_normal` in the second one - @seberg helped clarify. I agree with the original take, a PR to generalize `multivariate_normal` would be welcome.

As an aside, I must admit I had never heard of the "Swiss roll", but [scikit-learn's docs and implementation](https://github.com/scikit-learn/scikit-learn/blob/b194674c4/sklearn/datasets/_samples_generator.py#L1326) was very helpful.

**Comment by NeilGirdhar** [comment]
Another distribution that suffers from this limitation is `scipy.stats.dirichlet`.

**Comment by bashtage** [comment]
`randomgen.ExtendedGenerator.multivariate_normal` supports broadcasting mean and covariance.  I'll submit this to NumPy for 1.20.

https://bashtage.github.io/randomgen/generated/randomgen.generator.ExtendedGenerator.multivariate_normal.html#randomgen.generator.ExtendedGenerator.multivariate_normal
